src/syft/core/common/serde/deserialize.py

```diff
@@ -58,8 +58,9 @@ def _deserialize(
                 "provide a schematic to be filled with the hex data provided."
             )
         schematic = schema_type()
-        schematic.ParseFromString(bytes.fromhex(blob))
-        blob = schematic
+        if type(blob) == str:
```

Comment:
NITPICK:
```if type(blob) is str```

Reply:
Yeah i wondered about that, again it wasn't a warning so the question is.... is the type actually the same pointer in memory or is it just equal to the type. Are there going to be some variation in str types at some point which are == but are not "is / ===", not sure but is is probably more explicit, happy to change this as well.

Reply:
Interestingly, I have the "--strict-equality" flag already on, and mypy doesn't consider these to be incorrect.
I also don't think that the "type" can be coalesced into anything else, unlike say: 1.0 == 1